android/src/main/kotlin/vn/hunghd/flutterdownloader/DownloadWorker.kt

```diff
@@ -497,17 +497,33 @@ class DownloadWorker(context: Context, params: WorkerParameters) :
     private fun createFileInAppSpecificDir(filename: String, savedDir: String): File? {
         val newFile = File(savedDir, filename)
         try {
-            val rs: Boolean = newFile.createNewFile()
-            if (rs) {
-                return newFile
-            } else {
-                logError("It looks like you are trying to save file in public storage but not setting 'saveInPublicStorage' to 'true'")
+            if (newFile.exists()) {
+                val deleted = newFile.delete()
+                if (!deleted) {
+                    logError("Unable to delete existing file: ${newFile.absolutePath}")
+                    return null
+                }
+            }
```

Comment:
> I don't think it's expected to delete the old file to make place for the the newly downloaded file.
What should be done in this case? `fileName` passed in [`createFileInAppSpecificDir(String, String)`](https://github.com/fluttercommunity/flutter_downloader/blob/cc62a526119266e83c825f11b20c0dc00ea9ebd5/android/src/main/kotlin/vn/hunghd/flutterdownloader/DownloadWorker.kt#L497) already exists. Can we create a file with different (randomised index) name and return it? I think it will cause issues upstream from this function.

> Can we modify the error message to state not setting saveInPublicStorage is one of the possible causes, not the cause?

Yes, I wrote it as a question to the user (in the commit):
> Unable to create new file: ${newFile.absolutePath}.
> Are are trying to save file in public storage but not setting 'saveInPublicStorage' to 'true'?

This can definitely be updated to state it differently.

Reply:
> Yes, I wrote it as a question to the user (in the commit):

Yes, but the developer sees this log after the original file is already deleted.

Please also see [this issue](https://github.com/fluttercommunity/flutter_downloader/pull/708), I remember some agreement was reached in that issue what should we do in that case. But it's kinda abandoned, feel free to continue their work if you have time and will :)